sql/server.go

```diff
@@ -160,11 +160,12 @@ func (s *Server) exec(req driver.Request) (resp driver.Response, err error) {
 	// it is guaranteed not to be empty.
 	planMaker := planner{user: req.GetUser()}
 	defer func() {
-		// Update session state.
-		var marshalErr error
-		if resp.Session, marshalErr = gogoproto.Marshal(&planMaker.session); marshalErr != nil {
-			// Marshaling an internal protobuf can never fail.
-			panic(marshalErr)
+		// Update session state even if an error occurs.
+		if bytes, err := gogoproto.Marshal(&planMaker.session); err != nil {
+			// Marshaling a `Session` never errors (known from reading the code).
```

Comment:
note this is not necessarily true of all protos; i've only validated it for `Session`.